pandas/tests/indexing/multiindex/test_multiindex.py

```diff
@@ -207,6 +210,24 @@ def test_multiindex_with_na_missing_key(self):
         with pytest.raises(KeyError, match="missing_key"):
             df[[("missing_key",)]]
 
+    def test_multiindex_dtype_preservation(self):
+        # GH51261
+        columns = MultiIndex.from_tuples([("A", "B")], names=["lvl1", "lvl2"])
+        df = DataFrame(["value"], columns=columns).astype("category")
+        df_no_multiindex = df["A"]
+        assert is_categorical_dtype(df_no_multiindex["B"])
```

Comment:
```suggestion
        assert isinstance(df_no_multiindex["B"].dtype, CategoricalDtype)
```

`CategoricalDtype` will need an import at the top